mobly/snippet/client_base.py

```diff
@@ -127,54 +116,42 @@ def start_server(self):
       errors.ServerStartError: when failed to start the snippet server.
     """
 
-    @contextlib.contextmanager
-    def _execute_one_stage(stage):
-      """Context manager for executing one stage.
-
-      Args:
-        stage: StartServerStages, the stage which is running under this
-          context manager.
-
-      Yields:
-        None.
-      """
-      self.log.debug('[START_SERVER] Running the stage %s.', stage.name)
-      yield
-      self.log.debug('[START_SERVER] Finished the stage %s.', stage.name)
-
-    self.log.debug('Starting the server.')
+    self.log.debug('Initializing the snippet package %s.', self.package)
     start_time = time.perf_counter()
 
-    with _execute_one_stage(StartServerStages.BEFORE_STARTING_SERVER):
-      self.before_starting_server()
+    self.log.debug('Preparing starting the snippet server for %s.',
+                   self.package)
+    self.prepare_starting_server()
 
     try:
-      with _execute_one_stage(StartServerStages.DO_START_SERVER):
-        self.do_start_server()
-
-      with _execute_one_stage(StartServerStages.BUILD_CONNECTION):
-        self._build_connection()
+      self.log.debug('Starting the snippet server for %s.', self.package)
+      self.start_server()
 
-      with _execute_one_stage(StartServerStages.AFTER_STARTING_SERVER):
-        self.after_starting_server()
+      self.log.debug(
+          'Initializing a connection with the snippet server for %s.',
+          self.package)
+      self._init_connection()
 
     except Exception:
-      self.log.error('[START SERVER] Error occurs when starting the server.')
+      self.log.error(
+          'Error occurs when initializing the snippet server for %s.',
+          self.package)
       try:
         self.stop_server()
       except Exception:  # pylint: disable=broad-except
         # Only prints this exception and re-raises the original exception
-        self.log.exception('[START_SERVER] Failed to stop server because of '
-                           'new exception.')
+        self.log.exception(
+            'Failed to stop server for %s because of new exception.',
+            self.package)
 
       raise
 
-    self.log.debug('Snippet %s started after %.1fs on host port %d.',
+    self.log.debug('Snippet %s initialized after %.1fs on host port %d.',
                    self.package,
                    time.perf_counter() - start_time, self.host_port)
 
   @abc.abstractmethod
-  def before_starting_server(self):
+  def prepare_starting_server(self):
     """Prepares for starting the server.
```

Comment:
I think calling this `before_starting_server` is fine.
The docstring should align with the name.

You could further explain that these are prep steps that should happen before launching the snippet server on the remote side.

Reply:
Reverted it to `before_starting_server` and modified the docstring to explain.